sql/src/main/java/io/crate/operation/udf/JavaScriptUserDefinedFunction.java

```diff
@@ -132,6 +134,46 @@ private Object evaluateFunctionWithBindings(Bindings bindings, Input<Object>[] v
         return returnType.value(result);
     }
 
+    private Object processBytesRefInputIfNeeded(Object value) {
+        if (value instanceof BytesRef) {
+            value = BytesRefs.toString(value);
+        } else if (value instanceof Map) {
+            convertBytesRefToStringInMap(value);
+        } else if (value instanceof Object[]) {
+            convertBytesRefToStringInList(value);
+        }
+        return value;
+    }
+
+    private void convertBytesRefToStringInMap(Object value) {
+        if (value instanceof Map) {
```

Comment:
instance checks are already done outside - would move the cast there as well and only accept a Map as argument here